tests/clients/echo_client.py

```diff
@@ -0,0 +1,189 @@
+	except pconfig.ConfigurationException as e :
+		logger.error(str(e))
+		sys.exit(-1)
+
+	global direct_wrapper
+	direct_wrapper = DirectAdaptorFactoryWrapper(conffiles[0])
+
+	# Whether or not to connect to the registry list on the blockchain
+	off_chain = False
+
+	if options.registry_list:
+		config["ethereum"]["direct_registry_contract_address"] = \
+			options.registry_list
+
+	if options.service_uri:
+		config["tcf"]["json_rpc_uri"] = options.service_uri
+		off_chain = True
+
+	if options.off_chain:
+		off_chain = True
+
+	worker_id = options.worker_id
+	message = options.message
+	if options.message == None or options.message == "":
+		message = "Test Message"
+
+	# Initializing Worker Object
+	worker_obj = worker.WorkerDetails()
+
+def Main(args=None):
+	ParseCommandLine(args)
+
+	config["Logging"] = {
+		"LogFile" : "__screen__",
+		"LogLevel" : "INFO"
+	}
+	
+	plogger.setup_loggers(config.get("Logging", {}))
+	sys.stdout = plogger.stream_to_logger(
+		logging.getLogger("STDOUT"), logging.DEBUG)
+	sys.stderr = plogger.stream_to_logger(
+		logging.getLogger("STDERR"), logging.WARN)
+
+	logger.info("***************** INTEL TRUSTED COMPUTE FRAMEWORK (TCF)" + \
+		" *****************") 
+
+	# Connect to registry list and retrieve registry
+	if not off_chain:
+		direct_wrapper.init_worker_registry_list(config)
+		registry_lookup_result = direct_wrapper.registry_lookup()
+		if (registry_lookup_result[0] == 0):
+			logger.warn("No registries found")
+			sys.exit(1)
+		registry_retrieve_result = direct_wrapper.registry_retrieve(
+			registry_lookup_result[2][0])
+		config["tcf"]["json_rpc_uri"] = registry_retrieve_result[0]
+
+	# Prepare worker
+	direct_wrapper.init_worker_registry(config)
+
+	global worker_id
+	if not worker_id:
+		worker_lookup_json = jrpc_request.WorkerLookupJson(1, worker_type=1)
+		worker_lookup_result = direct_wrapper.worker_lookup(worker_lookup_json)
+		if "result" in worker_lookup_result and \
+			"ids" in worker_lookup_result["result"].keys():
+			if worker_lookup_result["result"]["totalCount"] != 0:
+				worker_id = worker_lookup_result["result"]["ids"][0]
+			else:
+				logger.error("ERROR: No workers found")
+				sys.exit(1)
+		else:
+			logger.error("ERROR: Failed to lookup worker")
+			sys.exit(1)
+
+	worker_retrieve_json = jrpc_request.WorkerRetrieveJson(2, worker_id)
+	worker_obj.load_worker(
+		direct_wrapper.worker_retrieve(worker_retrieve_json))
+
+	logger.info("**********Worker details Updated with Worker ID" + \
+		"*********\n%s\n", worker_id)
+
+	# Create, sign, and submit workorder 
+	# Convert workloadId to hex
+	workload_id = "echo-result"
+	workload_id = workload_id.encode("UTF-8").hex()
+	wo_submit_json = jrpc_request.WorkOrderSubmitJson(3, 6000, "pformat",
+		worker_id, workload_id, "0x2345", 
+		worker_encryption_key=worker_obj.worker_encryption_key, 
+		data_encryption_algorithm="AES-GCM-256")
+	wo_id = wo_submit_json.get_work_order_id()
+	wo_submit_json.add_in_data("bcde1111", message)
```

Comment:
Data hash has to be calculated. We shouldn't hard code.  Since this is optional field, you can use empty string or calculate it using supported hashing algorithm.